### DISCUSSION TITLE
UA Expert | opcua-asyncio vs. python-opcua

### DISCUSSION THREAD

**Discussion by dev-chris13** [FreeOpcUa/opcua-asyncio#714] [Q&A] [answered]
Hi!

I am trying to set up a small OPC UA server. I started by looking at the examples. I am using the UA Expert as a client.

I am not able to read attributes with  opcua-asyncio. Value and DataType are missing. I received the following error message:
`Read attributes of node NS2|Numeric|3 failed [ret = BadDecodingError]`. 
![image](https://user-images.githubusercontent.com/77881675/141191585-50b099f7-a8f2-416d-ac5c-d47bb6f77fd9.png)

When trying to call a function I get the messages
`Read result 0 returned status 'Good'` and `Read result 1 returned status 'Good'` and the method plugin GUI is incomplete. There is no option to define the input values.
![image](https://user-images.githubusercontent.com/77881675/141191640-15fb31e4-0b2f-4390-a80a-2fbcff57a3fb.png)

This is the GUI when using the same example from python-opcua:
![image](https://user-images.githubusercontent.com/77881675/141191847-f0ebbe92-fe6e-4055-8ec2-ad032c0657a9.png)

![image](https://user-images.githubusercontent.com/77881675/141191494-51ebc2a6-375a-419b-a83b-962e642a9c84.png)

What am I doing wrong?
Thanks for your help!
Chris

**Comment by AndreasHeine** [comment]
hm! unfortunately i cant reproduce that with our examples... can you share code? and versions (Library and UAExpert)?

![grafik](https://user-images.githubusercontent.com/56362817/141283834-ca59de1a-eab3-41f6-9831-99c29f12c90b.png)

**Comment by oroulet** [comment]
I see we have introduced several issues lately. Trying to fix them

**Reply by AndreasHeine** [reply]
is it on master?

**Reply by oroulet** [reply]
Yes..that array stuff and something with datclasses at least

**Comment by dev-chris13** [comment]
Hi guys! 
Thanks for your help.

I am currently using these versions: 
UA Expert: 1.4.6 237
Libraries: 
asyncua                0.9.91
opcua                  0.98.13

Used this example:
https://github.com/FreeOpcUa/opcua-asyncio/blob/master/examples/server-methods.py
Just changed port and server name to distinguish between both libraries.

**Comment by oroulet** [comment, marked as answer]
Can you test master on github? I pushed a fix. at least I can connect to uaexpert now

**Reply by dev-chris13** [reply]
Tested master! Works fine for me. Thanks for the quick fix.

**Comment by oroulet** [comment]
thanks for the feedback, I made a new release. That was an ugly breakage.
